src/hooks/useDeviceInfo.js

```diff
@@ -0,0 +1,20 @@
+import { isServer } from '../utils'
+
+export function useDeviceInfo() {
+  if (isServer()) return {}
+  const userAgent = window.navigator.userAgent
+  const isAndroid = () => Boolean(userAgent.match(/Android/i))
+  const isIOS = () => Boolean(userAgent.match(/iPhone|iPad|iPod/i))
+  const isOpera = () => Boolean(userAgent.match(/Opera Mini/i))
+  const isWindows = () => Boolean(userAgent.match(/IEMobile/i))
```

Comment:
`isWindowsPhone`? Since it's not just Windows 🤓

Reply:
Good one!